src/ssh-config.ts

```diff
@@ -164,13 +164,21 @@ export default class SSHConfig extends Array<Line> {
   public compute(opts: string | MatchOptions): Record<string, string | string[]> {
     if (typeof opts === 'string') opts = { Host: opts }
 
+    let userInfo: { username: string }
+    try {
+      userInfo = os.userInfo()
+    } catch {
+      // os.userInfo() throws a SystemError if a user has no username or homedir.
```

Comment:
FYI: this comment isn't necessarily precise

namely this can happen as well if the user is a "remote" user (e.g. LDAP scenario)

Reply:
It was copied from the documentation of os.userInfo() 😂 I guess if the user is "remote", homedir won't be available as well?